backend/main.py

```diff
@@ -114,14 +114,23 @@ async def start_spotihue(lights: List[str] = None):
     if not lights:
         raise HTTPException(status_code=400, detail='"lights" list is required.')
 
-    available_lights = spotihue.retrieve_available_lights()
+    available_lights = [light['light_name'] for light in spotihue.retrieve_available_lights()]
     lights = [light for light in lights if light in available_lights]
 
     try:
-        # TODO: make this idempotent
+        spotihue_task_running = tasks.is_spotihue_running()
 
-        task = tasks.run_spotihue.delay(lights)
-        redis_client.set("spotihue", str(task.id))
+        if spotihue_task_running:
+            logger.info('Spotihue is already running')
+        else:
+            forget_spotihue = tasks.clear_spotihue_task_id.signature()
```

Comment:
What exactly is this doing?

Reply:
I noticed that if the task ends via some manner other than `stop_spotihue()` terminating it (like if 1. the user has no currently-playing Spotify track - even a paused one - OR 2. if some other unlikely error were to be thrown), then the cached "spotihue_task_id" remains cached in Redis (even though the task isn't running anymore). This just makes extra sure that the cached task ID gets deleted once the task is over (whether through the task's 1. link param, 2. its link_error param, or 3. by stop_spotihue() deleting it).
We can also get this "is spotihue running" boolean by looking at all ("all" == 1, in our case) of our running celery tasks, but since we technically have more than one celery task defined (and we could add more in the future), i liked the idea of querying directly for 1 task ID a little better.

Reply:
Woops, i didn't read your question correctly lol - the `signature()` method turns the normal Python function task into a celery [Signature](https://docs.celeryq.dev/en/stable/reference/celery.html#celery.Signature) object. I guess apply_async() does that for us under the hood when we call `task_function.apply_async()`, but not when we just pass the a `task_function` in as a function param...